conformance/ipv6.go

```diff
@@ -0,0 +1,196 @@
+package conformance
+
+import (
+	"encoding/json"
+	"time"
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/sriov-tests/pkg/util/cluster"
+	"github.com/openshift/sriov-tests/pkg/util/execute"
+	"github.com/openshift/sriov-tests/pkg/util/namespaces"
+
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const testNamespace = "sriov-testing"
+const testerImage = "centos:centos8"
+const sriovInterfaceName = "net1"
+const numVfs = 5
+const resourceName = "sriovnic"
+const ipv6NetworkName = "ipv6network"
+const hostnameLabel = "kubernetes.io/hostname"
+
+// Needed for parsing of podinfo
+type Network struct {
+	Interface string
+	Ips       []string
+}
+
+var _ = FDescribe("sriov", func() {
+	var sriovInfos *cluster.EnabledNodes
+	var testNode string
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(testNamespace, clients)
+		Expect(err).ToNot(HaveOccurred())
+
+		sriovInfos, err = cluster.DiscoverSriov(clients, operatorNamespace)
+		Expect(err).ToNot(HaveOccurred())
+		testNode = sriovInfos.Nodes[0]
+	})
+
+	BeforeEach(func() {
+		err := namespaces.Clean(operatorNamespace, testNamespace, clients)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			res, err := cluster.SriovStable(operatorNamespace, clients)
+			Expect(err).ToNot(HaveOccurred())
+			return res
+		}, 3*time.Minute, 1*time.Second).Should(Equal(true))
+	})
+
+	Describe("sriov", func() {
+		var _ = Context("IPv6 configured secondary interfaces on pods", func() {
+			It("should be able to ping each other", func() {
+				sriovDevice, err := sriovInfos.FindOneSriovDevice(testNode)
+				Expect(err).ToNot(HaveOccurred())
+				createSriovPolicy(testNode, sriovDevice.Name, testNode)
+				createSriovNetwork()
+			
+				pod := createTestPod([]string{"/bin/bash", "-c", "--"},
+					[]string{"while true; do sleep 300000; done;"}, testNode, ipv6NetworkName)
+				ips := getSriovNicIps(pod)
+				Expect(ips).NotTo(BeNil(), "No sriov network interface found.")
+				Expect(len(ips)).Should(Equal(1))
+				for _, ip := range ips {
+					pingPod(ip, testNode, ipv6NetworkName)
+				}
+			})
+		})
+	})
+})
+
+func createSriovNetwork() {
+	sriovNetwork := &sriovv1.SriovNetwork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ipv6NetworkName,
+			Namespace: operatorNamespace,
+		},
+		Spec: sriovv1.SriovNetworkSpec{
+			ResourceName:     resourceName,
+			IPAM:             `{"type": "host-local","ranges": [[{"subnet": "3ffe:ffff:0:01ff::/64"}]],"dataDir": "/run/my-orchestrator/container-ipam-state"}`,
```

Comment:
after you move the function please make this one a variable for the function

Reply:
Done